src/utils/string/StringUtils.js

```diff
@@ -477,11 +477,14 @@ const getCaptionWithTopicStyle = (
   const id = removePrefixTopic(topicWithPrefix);
   const topicRegex = /\B(\#[a-zA-Z0-9_+-]+\b)(?!;)/;
   const validationTextHasAt = /\B(\@[a-zA-Z0-9_+-]+\b)(?!;)/;
+  const urlRegex = /^((?:https?:\/\/)?[^.\/]+(?:\.[^.\/]+)+(?:\/.*)?)(\s*.*)?$/;
+
   if (substringEnd && typeof substringEnd === 'number') {
     text = text.substring(0, substringEnd);
   } else {
     text = text.substring(0, text.length);
   }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The introduction of `urlRegex` is good for detecting URLs in the text. However, the assignment of `text` in both branches of the conditional statement seems redundant as it does the same thing. The else branch can be removed to simplify the code.
```diff
-    if (substringEnd && typeof substringEnd === 'number') {
-      text = text.substring(0, substringEnd);
-    } else {
-      text = text.substring(0, text.length);
-    }
+    if (substringEnd && typeof substringEnd === 'number') {
+      text = text.substring(0, substringEnd);
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->